agent/engine/data.go

```diff
@@ -220,6 +243,21 @@ func (engine *DockerTaskEngine) removeTaskData(task *apitask.Task) {
 			seelog.Errorf("Failed to remove data for container %s: %v", c.Name, err)
 		}
 	}
+	containers, ok := engine.state.PulledContainerMapByArn(task.Arn)
+	if ok {
```

Comment:
nit: not necessary to check for `ok==true`, you can just iterate over containers, even if it's nil it won't cause errors